### PR TITLE
doc/rados: edit troubleshooting-pg.rst

### PR DESCRIPTION
**Pull request by zdover23**
Edit doc/rados/troubleshooting/troubleshooting-pg.rst (1 of x).

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: doc/rados/troubleshooting/troubleshooting-pg.rst [hunk 1]
```diff
@@ -1,120 +1,128 @@
...
-=====================
+====================
 
 Placement Groups Never Get Clean
 ================================
 
-When you create a cluster and your cluster remains in ``active``,
-``active+remapped`` or ``active+degraded`` status and never achieves an
-``active+clean`` status, you likely have a problem with your configuration.
+If, after you have created your cluster, it remains in `the `active`` status,
```
Comment (anthonyeleven) [inline doc/rados/troubleshooting/troubleshooting-pg.rst:8]: , any Placement Groups (PGs) remain in

File: doc/rados/troubleshooting/troubleshooting-pg.rst [hunk 2]
```diff
@@ -1,120 +1,128 @@
...
+the ``active+remapped`` status or the ``active+degraded`` status and never
+achieves an ``active+clean`` status, you likely have a problem with your
+configuration.
 
-You may need to review settings in the `Pool, PG and CRUSH Config Reference`_
-and make appropriate adjustments.
+In such a situation ,it may be necessary to review settings in the `Pool, PG
```
Comment (anthonyeleven) [inline doc/rados/troubleshooting/troubleshooting-pg.rst:13]: review the settings

File: doc/rados/troubleshooting/troubleshooting-pg.rst [hunk 3]
```diff
@@ -1,120 +1,128 @@
...
+and CRUSH Config Reference`_ and make appropriate adjustments.
 
-As a general rule, you should run your cluster with more than one OSD and a
-pool size greater than 1 object replica.
+As a general rule, run your cluster with more than one OSD and a pool size
+greater than one object replica.
```
Comment (anthonyeleven) [inline doc/rados/troubleshooting/troubleshooting-pg.rst:17]: Let's advise greater than two.  Some people -- including people who should know better -- think 2R is just dandy.

File: doc/rados/troubleshooting/troubleshooting-pg.rst [hunk 4]
```diff
@@ -1,120 +1,128 @@
...
-Ceph no longer provides documentation for operating on a single node, because
-you would never deploy a system designed for distributed computing on a single
-node. Additionally, mounting client kernel modules on a single node containing a
-Ceph  daemon may cause a deadlock due to issues with the Linux kernel itself
-(unless you use VMs for the clients). You can experiment with Ceph in a 1-node
+Ceph no longer provides documentation for operating on a single node.  Systems
+designed for distributed computing by definition do not run on a single node.
+The mounting of client kernel modules on a single node that contains a Ceph
+daemon may cause a deadlock due to issues with the Linux kernel itself (unless
+VMs are used as clients). You can experiment with Ceph in a 1-node
```
Comment (anthonyeleven) [inline doc/rados/troubleshooting/troubleshooting-pg.rst:28]: one

File: doc/rados/troubleshooting/troubleshooting-pg.rst [hunk 5]
```diff
@@ -1,120 +1,128 @@
...
 
-If you are trying to create a cluster on a single node, you must change the
-default of the ``osd_crush_chooseleaf_type`` setting from ``1`` (meaning
-``host`` or ``node``) to ``0`` (meaning ``osd``) in your Ceph configuration
-file before you create your monitors and OSDs. This tells Ceph that an OSD
-can peer with another OSD on the same host. If you are trying to set up a
-1-node cluster and ``osd_crush_chooseleaf_type`` is greater than ``0``,
-Ceph will try to peer the PGs of one OSD with the PGs of another OSD on
-another node, chassis, rack, row, or even datacenter depending on the setting.
+To create a cluster on a single node, you must change the default of the
```
Comment (anthonyeleven) [inline doc/rados/troubleshooting/troubleshooting-pg.rst:31]: change the blah setting from the default of ``1``

File: doc/rados/troubleshooting/troubleshooting-pg.rst [hunk 6]
```diff
@@ -1,120 +1,128 @@
...
+``osd_crush_chooseleaf_type`` setting from ``1`` (meaning ``host`` or ``node``)
+to ``0`` (meaning ``osd``) in your Ceph configuration file before you create
+your monitors and OSDs. This tells Ceph that an OSD is permitted to peer with
```
Comment (anthonyeleven) [inline doc/rados/troubleshooting/troubleshooting-pg.rst:34]: Here and below, s/peer/place/

**Review by anthonyeleven — APPROVED** [review]

File: doc/rados/troubleshooting/troubleshooting-pg.rst (hunk 1 above)
Comment (zdover23) [reply doc/rados/troubleshooting/troubleshooting-pg.rst:8]: Accepted.

File: doc/rados/troubleshooting/troubleshooting-pg.rst (hunk 3 above)
Comment (zdover23) [reply doc/rados/troubleshooting/troubleshooting-pg.rst:17]: Accepted.

File: doc/rados/troubleshooting/troubleshooting-pg.rst (hunk 5 above)
Comment (zdover23) [reply doc/rados/troubleshooting/troubleshooting-pg.rst:31]: Accepted.

File: doc/rados/troubleshooting/troubleshooting-pg.rst [hunk 7]
```diff
@@ -1,120 +1,128 @@
...
 -------------
 
-If you have the ``osd_pool_default_size`` set to ``1``, you will only have
-one copy of the object. OSDs rely on other OSDs to tell them which objects
-they should have. If a first OSD has a copy of an object and there is no
-second copy, then no second OSD can tell the first OSD that it should have
-that copy. For each placement group mapped to the first OSD (see
-``ceph pg dump``), you can force the first OSD to notice the placement groups
-it needs by running::
+If you have the ``osd_pool_default_size`` set to ``1``, you will have only one
```
[the remark this answers is not in the document]
Comment (zdover23) [reply doc/rados/troubleshooting/troubleshooting-pg.rst:71]: Yeah, that is more correct.

File: doc/rados/troubleshooting/troubleshooting-pg.rst (hunk 2 above)
Comment (zdover23) [reply doc/rados/troubleshooting/troubleshooting-pg.rst:13]: Accepted.

File: doc/rados/troubleshooting/troubleshooting-pg.rst [hunk 8]
```diff
@@ -1,120 +1,128 @@
...
+.. prompt:: bash
+
+   ceph osd force-create-pg <pgid>
 
 
 CRUSH Map Errors
 ----------------
 
-Another candidate for placement groups remaining unclean involves errors
+If the placement groups in your cluster are unclean, then there might be errors
```
[the remark this answers is not in the document]
Comment (zdover23) [reply doc/rados/troubleshooting/troubleshooting-pg.rst:87]: Accepted.

File: doc/rados/troubleshooting/troubleshooting-pg.rst [hunk 9]
```diff
@@ -1,120 +1,128 @@
...
 ======================
 
-It is normal for placement groups to enter states like "degraded" or "peering"
-following a failure.  Normally these states indicate the normal progression
-through the failure recovery process. However, if a placement group stays in one
-of these states for a long time this may be an indication of a larger problem.
-For this reason, the monitor will warn when placement groups get "stuck" in a
-non-optimal state.  Specifically, we check for:
+It is normal for placement groups to enter "degraded" or "peering" states after
+a failure. Normally, these states indicate the expected progression through the
```
[the remark this answers is not in the document]
Comment (zdover23) [reply doc/rados/troubleshooting/troubleshooting-pg.rst:95]: accepted.

File: doc/rados/troubleshooting/troubleshooting-pg.rst [hunk 10]
```diff
@@ -1,120 +1,128 @@
...
+failure recovery process. However, a placement group that stays in one of these
+states for a long time might be an indication of a larger problem. For this
+reason, the monitor will warn when placement groups get "stuck" in a
```
[the remark this answers is not in the document]
Comment (zdover23) [reply doc/rados/troubleshooting/troubleshooting-pg.rst:98]: Accepted.

File: doc/rados/troubleshooting/troubleshooting-pg.rst [hunk 11]
```diff
@@ -188,32 +206,39 @@ Recovery will proceed.
...
 * Before the new objects are copied, 2 goes down.
 
-Now 1 knows that these object exist, but there is no live ``ceph-osd`` who
-has a copy.  In this case, IO to those objects will block, and the
-cluster will hope that the failed node comes back soon; this is
-assumed to be preferable to returning an IO error to the user.
+At this point, 1 knows that these objects exist, but there is no live
+``ceph-osd`` that has a copy of the objects. In this case, IO to those objects
+will block, and the cluster will hope that the failed node comes back soon.
+This is assumed to be preferable to returning an IO error to the user.
```
[the remark this answers is not in the document]
Comment (zdover23) [reply doc/rados/troubleshooting/troubleshooting-pg.rst:235]: I have done so.

File: doc/rados/troubleshooting/troubleshooting-pg.rst [hunk 12]
```diff
@@ -289,31 +316,31 @@ data, but it is ``down``.  The full range of possible states include:
...
 locations.
 
-It is possible that there are other locations where the object can
-exist that are not listed.  For example, if a ceph-osd is stopped and
-taken out of the cluster, the cluster fully recovers, and due to some
-future set of failures ends up with an unfound object, it won't
-consider the long-departed ceph-osd as a potential location to
-consider.  (This scenario, however, is unlikely.)
+It is possible that there are other locations where the object can exist that
+are not listed. For example: if a ceph-osd is stopped and taken out of the
```
[the remark this answers is not in the document]
Comment (zdover23) [reply doc/rados/troubleshooting/troubleshooting-pg.rst:320]: Accepted.

File: doc/rados/troubleshooting/troubleshooting-pg.rst [hunk 13]
```diff
@@ -289,31 +316,31 @@ data, but it is ``down``.  The full range of possible states include:
...
+cluster and then the cluster fully recovers, and then through a subsequent set
+of failures the cluster ends up with an unfound object, the cluster will ignore
+the removed ceph-osd. (This scenario, however, is unlikely.)
```
[the remark this answers is not in the document]
Comment (zdover23) [reply doc/rados/troubleshooting/troubleshooting-pg.rst:323]: Accepted.

**Pull request merged**